crm_salesperson_planner_re_assign/models/crm_salesperson_planner_visit_reassign.py

```diff
@@ -0,0 +1,208 @@
+        if self.template_draft_state:
+            template_states.append("draft")
+        if self.template_in_progress_state:
+            template_states.append("in-progress")
+        if self.template_done_state:
+            template_states.append("done")
+        if self.template_cancel_state:
+            template_states.append("cancel")
+        return template_states
+
+    def get_visit_states(self):
+        visit_states = []
+        if self.visit_draft_state:
+            visit_states.append("draft")
+        if self.visit_confirm_state:
+            visit_states.append("confirm")
+        if self.visit_done_state:
+            visit_states.append("done")
+        if self.visit_cancel_state:
+            visit_states.append("cancel")
+        if self.visit_incident_state:
+            visit_states.append("incident")
+        return visit_states
+
+    def action_search(self):
+        self.salesperson_reassign_line_ids.unlink()
+        self.visit_reassign_line_ids.unlink()
+        lines = []
+        if self.search_templates:
+            template_states = self.get_template_states()
+            if template_states:
+                template_ids = self.env[
+                    "crm.salesperson.planner.visit.template"
+                ].search(
+                    [
+                        ("user_id", "=", self.current_user_id.id),
+                        ("state", "in", template_states),
+                    ]
+                )
+                if template_ids:
+                    for t in template_ids:
+                        lines.append(
+                            {
+                                "salesperson_visit_template_reassign_id": self.id,
+                                "salesperson_visit_template_id": t.id,
+                            }
+                        )
+        if self.search_visits:
+            visit_states = self.get_visit_states()
+            if visit_states:
+                visits_search = [
+                    ("user_id", "=", self.current_user_id.id),
+                    ("state", "in", visit_states),
+                ]
+                if self.start_date:
+                    visits_search.append(("date", ">=", self.start_date))
+                if self.end_date:
+                    visits_search.append(("date", "<=", self.end_date))
+                visit_ids = self.env["crm.salesperson.planner.visit"].search(
+                    visits_search
+                )
+                if visit_ids:
+                    for v in visit_ids:
+                        lines.append(
+                            {
+                                "salesperson_visit_reassign_id": self.id,
+                                "crm_salesperson_planner_visit_id": v.id,
+                            }
+                        )
+        if lines:
+            self.env["crm.salesperson.planner.visit.reassign.line"].create(lines)
+
+    def action_validate(self):
+        if not self.new_user_id:
```

Comment:
Why not doing it required instead?

Reply:
It is not required as we'd like to give the chance of creating reassignation templates before knowing who the new salesperson will be.